### PR TITLE
Possibly fixes #1422 (event queue overflow)

### PR DESCRIPTION
**Pull request by hostilefork**
Event list expansion was added by Atronix to deal with bug #1422.
Original expansion set EVENT_LIMIT to 0xFFFF REB_EVENT values (64k).
This was much larger than the hard limit of 128 set by original
open-source Rebol.  But adding `SET_SIGNAL(SIG_EVENT_PORT)` in
`Append_Event()` corrected the bug and allowed the limit to be
lowered.  Still the chunking/expansion mechanic is retained.

### CODE REVIEW

File: src/core/c-do.c [hunk 1]
```diff
@@ -446,12 +446,12 @@ void Trace_Arg(REBINT num, const REBVAL *arg, const REBVAL *path)
...
-#ifdef NOT_USED_INVESTIGATE
-	if (GET_FLAG(sigs, SIG_EVENT_PORT)) {  // !!! Why not used?
+	// Was #ifdef'd out in R3-Alpha open source code. Enabled due to #1422
+	if (GET_FLAG(sigs, SIG_EVENT_PORT)) {
+		REBINT result = Awake_System(NULL, FALSE); // !!! just event port?
 		CLR_SIGNAL(SIG_EVENT_PORT);
-		Awake_Event_Port();
+		cast(void, result); // !!! Should the result be acted on?
 	}
-#endif
```
Comment (zsx) [inline src/core/c-do.c]: This will cause unexpected recursions. Think about a button with an action function. When the button is clicked, the action function could call Do_Singals and then Awake_System, which handles the pending events. If there are some  pending clicks on this button, the same action function will then be called and cause recursive calls. This is similar to the recursion I wanted to avoid with wait/only.

Basically, we'd like the Awake_System is only be called from WAIT, so that the user knows there is only one place that event handling could happen.

Comment (hostilefork) [inline src/core/c-do.c]: Hmmm... I see.  But perhaps should it be not just WAIT, but also at sort of the "top level loop" (e.g. if you get back to the REPL)?

**Comment by hostilefork** [comment]
Closing because this apparently doesn't completely solve the problem, and because Ren-C is aiming to become agnostic to the event system used by the host.  Hence this is not a priority concern.
